### ISSUE TITLE
Vault provider wrongly using child namespace name in provider config in 'sys/seal-status' call

### ISSUE THREAD

**Issue by faseyiks** [hashicorp/terraform-provider-vault#1948]
### Terraform Version
terraform.exe version
Terraform v1.5.3
on windows_amd64
+ provider registry.terraform.io/hashicorp/vault v3.18.0

This is also happening with v3.17.0 and v3.16.0

### Affected Resource(s)
Please list the resources as a list, for example:
- provider config

If this issue appears to affect multiple resources, it may be an issue with Terraform's core, so please mention this.

### Terraform Configuration Files
```hcl
provider "vault" {
  address      = var.vault_addr
  token        = var.token
  alias        = "parent_namespace"
  namespace    = var.parent_namespace_name
  ca_cert_file = var.ca_cert_file
}

provider "vault" {
  address      = var.vault_addr
  token        = var.token
  alias        = "child_namespace"
  namespace    = format("%s/%s", var.parent_namespace_name, var.child_namespace_name)
  ca_cert_file = var.ca_cert_file
}
```

### Expected Behavior
When making a call to the `sys/seal-status` endpoint, it should not set `X-Vault-Namespace` in `sys/seal-status` call.

### Actual Behavior
 `X-Vault-Namespace` is set against the provider.

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Typical error we have seen in our environment is. The `no handler for route` error is of course correct because the sys/seal-status should never be called against the child namespace and in this case a child namepsace that has not even been created or may not have been created yet.
```
╷
│ Error: could not determine the Vault server version, err=Error making API request.
│ 
│ Namespace: PARENT_NS/CHILD_NS
│ URL: GET https://<VAULT_ADDR>:8200/v1/sys/seal-status
│ Code: 404. Errors:
│
│ * no handler for route "<CHILD_NS>/sys/seal-status". route entry not found.
│
│   with module.vault_namespace.provider["registry.terraform.io/hashicorp/vault"].new_namespace,
│   on .terraform\modules\vault_namespace\main.tf line 20, in provider "vault":
│   20: provider "vault" {
│
╵
```

**Comment by pjastrzabek** [comment]
I'm facing similar issue and found this article
https://support.hashicorp.com/hc/en-us/articles/19141667372947-Terraform-Vault-Provider-returns-Error-could-not-determine-the-Vault-server-version-err-Error-making-API-request-
where they suggest to pin provider to 3.15.0

**Comment by faseyiks** [comment]
We implemented the workaround and informed hashicorp about it.
However, we think the bug should be fixed sooner rather than later 
```
provider "vault" {
  address      = var.vault_addr
  token        = var.token
  alias        = "child_namespace"
  namespace    = format("%s/%s", var.parent_namespace_name, child_namespace_name.path_fq)
  ca_cert_file = var.ca_cert_file
}
```

**Comment by OneWhoNests** [comment]
This should be fixed in 3.20.

**Comment by fairclothjm** [comment]
Fixed in https://github.com/hashicorp/terraform-provider-vault/pull/1967
